archive/100-theorems-list/37_solution_of_cubic.lean

```diff
@@ -0,0 +1,199 @@
+/-
+Copyright (c) 2021 Jeoff Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeoff Lee
+-/
+import tactic.basic
+import data.complex.basic
+import algebra.group.defs
+
+/-!
+# The Solution of Cubic
+
+This file proves Theorem 37 from the [100 Theorems List](https://www.cs.ru.nl/~freek/100/).
+
+The theorem `cubic_eq_zero_iff` gives solution to the cubic equation over ℂ,
+`cubic_eq_zero_iff_of_discrim_eq_zero`
+based on the [Cardano's Formula](https://en.wikipedia.org/wiki/Cubic_equation#Cardano's_formula).
+
+## References
+
+Originally ported from Isabelle/HOL. The
+[original file](https://isabelle.in.tum.de/dist/library/HOL/HOL-ex/Cubic_Quartic.html) was written by Amine Chaieb.
+
+## Tags
+polynomial, cubic, root
+-/
+
+section
+
+open complex
+
+variables (a b c d : ℂ)
```

Comment:
Would this work over a general algebraically closed field?

Reply:
A field with char ≠ 2, 3, and having cube root of unity would suffice. the rest hypotheses `r^2=..., s^3=...` would do the work.

Reply:
currently trying to replace `ℂ` with `[field K] [invertible (2 : K)] [invertible (3 : K)]` and `{ω : ℂ} (hω : is_primitive_root ω 3)`